### PR TITLE
Improve various validations

### PR DESCRIPTION
**Pull request by howardjohn**
Just misc improvements.

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -547,19 +546,23 @@ func validateServer(server *networking.Server) (v Validation) {
...
 	}
 	if protocol.Parse(port.Protocol) == protocol.Unsupported {
-		errs = appendErrors(errs, fmt.Errorf("invalid protocol %q, supported protocols are HTTP, HTTP2, GRPC, GRPC-WEB, MONGO, REDIS, MYSQL, TCP", port.Protocol))
+		errs = appendValidation(errs, fmt.Errorf("invalid protocol %q, supported protocols are HTTP, HTTP2, GRPC, GRPC-WEB, MONGO, REDIS, MYSQL, TCP", port.Protocol))
 	}
-	if port.Number > 0 {
-		errs = appendErrors(errs, ValidatePort(int(port.Number)))
+	if port.Number > 0 || !strings.HasPrefix(bind, UnixAddressPrefix) {
+		errs = appendValidation(errs, ValidatePort(int(port.Number)))
+	}
```
Comment (howardjohn) [inline pkg/config/validation/validation.go:558]: this was valid but leads to NACK, so make it an error

File: pkg/config/validation/validation.go [hunk 2]
```diff
@@ -547,19 +546,23 @@ func validateServer(server *networking.Server) (v Validation) {
...
+	// nolint: staticcheck
+	if port.TargetPort > 0 {
+		errs = appendValidation(errs, fmt.Errorf("targetPort has no impact on Gateways"))
```
Comment (howardjohn) [inline pkg/config/validation/validation.go:561]: Add warning, as field does nothing

File: pkg/config/validation/validation.go [hunk 3]
```diff
@@ -3017,21 +3029,23 @@ func validateHTTPFaultInjection(fault *networking.HTTPFaultInjection) (errs erro
...
+		errs = appendValidation(errs, validateGRPCStatus(abort.GetGrpcStatus()))
 	case *networking.HTTPFaultInjection_Abort_Http2Error:
 		// TODO: HTTP2 error validation
-		errs = multierror.Append(errs, errors.New("HTTP/2 abort fault injection not supported yet"))
+		errs = appendValidation(errs, errors.New("HTTP/2 abort fault injection not supported yet"))
 	case *networking.HTTPFaultInjection_Abort_HttpStatus:
-		errs = appendErrors(errs, validateHTTPStatus(abort.GetHttpStatus()))
+		errs = appendValidation(errs, validateHTTPStatus(abort.GetHttpStatus()))
+	default:
+		errs = appendWarningf(errs, "abort configured, but error type not set")
```
Comment (howardjohn) [inline pkg/config/validation/validation.go:3048]: Warn if they have `abort: {}` which does nothing

**Review by ramaraochavali — APPROVED** [review]

**Comment by wulianglongrd** [comment]
> /retest

It looks like there is something wrong with the `/retest` command, it should only trigger failed jobs, but it actually triggers all jobs now.

**Pull request merged**